### PR TITLE
Add labs flag for Threads Activity Centre

### PR DESCRIPTION
**Pull request by florianduros**
## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

Content:
- Integrate https://github.com/matrix-org/matrix-react-sdk/pull/12156
- Add a labs flag for the future Threads Activity Centre

---
Here's what your changelog entry will look like:

## ✨ Features
 * Add labs flag for Threads Activity Centre ([\#12137](https://github.com/matrix-org/matrix-react-sdk/pull/12137)). Contributed by @florianduros.

### CODE REVIEW

File: src/RoomNotifs.ts [hunk 1]
```diff
@@ -81,9 +81,12 @@ export function setRoomNotifsState(client: MatrixClient, roomId: string, newStat
 }
 
 export function getUnreadNotificationCount(room: Room, type: NotificationCountType, threadId?: string): number {
-    let notificationCount = !!threadId
-        ? room.getThreadUnreadNotificationCount(threadId, type)
-        : room.getUnreadNotificationCount(type);
+    const countShownForRoom = (): number =>
+        SettingsStore.getValue("threadsActivityCentre")
```
Comment (t3chguy) [inline src/RoomNotifs.ts:85]: Settings isn't very performant, could we pass this value in and only read it periodically? I presume this method is called significantly

File: src/Unread.ts [hunk 2]
```diff
@@ -57,7 +57,12 @@ export function doesRoomHaveUnreadMessages(room: Room): boolean {
         return false;
     }
 
-    for (const withTimeline of [room, ...room.getThreads()]) {
+    const toCheck: Array<Room | Thread> = [room];
+    if (!SettingsStore.getValue("threadsActivityCentre")) {
```
Comment (t3chguy) [inline src/Unread.ts:61]: Same as above

**Review by t3chguy — DISMISSED** [review]

File: src/RoomNotifs.ts (hunk 1 above)
Comment (dbkr) [reply src/RoomNotifs.ts:85]: Hmm... kinda. I'm considering whether it would be simpler to memoise settings getters...

Comment (dbkr) [reply src/RoomNotifs.ts]: ok, done

**Review by dbkr — APPROVED** [review]

**Pull request merged**